app/code/Magento/Downloadable/Helper/File.php

```diff
@@ -134,7 +134,7 @@ protected function _moveFileFromTmp($baseTmpPath, $basePath, $file)
         $destFile = dirname(
             $file
         ) . '/' . \Magento\MediaStorage\Model\File\Uploader::getNewFileName(
-            $this->getFilePath($basePath, $file)
+            $this->getFilePath($this->_mediaDirectory->getAbsolutePath($basePath), $file)
```

Comment:
Hi @geet07. Could I kindly ask you to provide more information about your solution, please? According to the original issue, the problem is the destination path is the same for different downloadable products with the same file name. 

After this change, the destination path will be based on an absolute path to the file, but will still be the same for different downloadable products with the same file name, see [getBasePath()](https://github.com/magento/magento2/blob/b2ce2a37d921b5ad88fc38663fc0ff3dd6c582d1/app/code/Magento/Downloadable/Model/Link.php#L121)